### PR TITLE
Create merge_queue.md

### PR DESCRIPTION
**Pull request by Piinks**
Adds information on the merge queue, to be used for announcement. Will update to final doc after the merge project is complete.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [ ] I followed the [breaking change policy] and added [Data Driven Fixes] where supported.
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#overview
[Tree Hygiene]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md
[test-exempt]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/blob/main/docs/contributing/Style-guide-for-Flutter-repo.md
[Features we expect every widget to implement]: https://github.com/flutter/flutter/blob/main/docs/contributing/Style-guide-for-Flutter-repo.md#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/blob/main/docs/contributing/Chat.md
[Data Driven Fixes]: https://github.com/flutter/flutter/blob/main/docs/contributing/Data-driven-Fixes.md

### CODE REVIEW

**Comment by Piinks** [comment]
I need to check in the images quickly elsewhere. We have wiki image storage in flutter/assets-for-api-docs

File: docs/infra/merge_queue.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
+## Github Merge Queue
+
+If you are not hacking on flutter/flutter, you can stop reading now.
+
+tl;dr GitHub merge queue will be enabled in the flutter/flutter repo, which slightly changes the PR merge process, but you should not notice a change in your workflow.
+
+On Friday November 14, 2024, GitHub merge queue feature will be enabled in the flutter/flutter repo. After this change, pull requests will first need to pass merge queue tests before landing on the target branch (e.g. main or master). At this time, no additional tests will run in the merge queue. A PR that passes presubmit checks will be allowed to land on the target branch immediately.
```
Comment (yjbanov) [inline docs/infra/merge_queue.md:7]: It was pointed out to me that Friday is November 15 this year :)

**Review by yjbanov — APPROVED** [review]

File: docs/infra/merge_queue.md (hunk 1 above)
Comment (Piinks) [reply docs/infra/merge_queue.md:7]: TIL. 😜

File: docs/infra/merge_queue.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+What should I expect?
+The only visible change will be the merge button. Currently, the button looks like this:
+
+<!-- TODO(Piinks) Image -->
```
Comment (Piinks) [inline docs/infra/merge_queue.md:11]:
```suggestion
![Squash and merge button](https://github.com/flutter/assets-for-api-docs/blob/main/assets/wiki/squash_merge.png)
```

File: docs/infra/merge_queue.md [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+With the merge queue, the button will now look like this:
+
+<!-- TODO(Piinks) Image -->
```
Comment (Piinks) [inline docs/infra/merge_queue.md:15]:
```suggestion
![Merge when ready button](https://github.com/flutter/assets-for-api-docs/blob/main/assets/wiki/merge_ready.png)
```

File: docs/infra/merge_queue.md [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+
+Unlike "Squash and merge", the new button does not immediately land the PR on the target branch, but instead enqueues your PR for extra checks in the merge queue. If you do not do anything, eventually your PR will land.
+Can I still use autosubmit?
+Yes! Feel free to continue using the autosubmit label as normal. The bots will run your PR through the merge queue and land it automatically.
+What's "Remove from queue"? Can I click it?
+While your PR is enqueued, Github will show the following button:
+
+<!-- TODO(Piinks) Image -->
```
Comment (Piinks) [inline docs/infra/merge_queue.md:24]:
```suggestion
![Remove from queue button](https://github.com/flutter/assets-for-api-docs/blob/main/assets/wiki/remove_queue.png)
```

File: docs/infra/merge_queue.md (hunk 1 above)
Comment (Piinks) [inline docs/infra/merge_queue.md:7]:
```suggestion
On Friday November 15, 2024, GitHub merge queue feature will be enabled in the flutter/flutter repo. After this change, pull requests will first need to pass merge queue tests before landing on the target branch (e.g. main or master). At this time, no additional tests will run in the merge queue. A PR that passes presubmit checks will be allowed to land on the target branch immediately.
```

File: docs/infra/merge_queue.md [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+What's "Remove from queue"? Can I click it?
+While your PR is enqueued, Github will show the following button:
+
+![Remove from queue button](https://github.com/flutter/assets-for-api-docs/blob/main/assets/wiki/remove_queue.png)
+
+When you press this button, GitHub will remove your PR from the queue, and the PR will no longer be landed by any automated machinery.
+
+WARNING: please only use this button in case of emergency. Adding and removing your PR from the queue costs in CI resources. Any CI work being done for your PR and all PRs that follow yours will be stopped and restarted.
+What if I have questions?
+If you have any questions or concerns, or If you suspect you are unable to land your PRs for reasons other than the familiar test failures and flakes, please contact any of: yjbanov@google.com, codefu@google.com, fujino@google.com, katelovett@google.com.
```
Comment (Piinks) [inline docs/infra/merge_queue.md:30]:
```suggestion
If you have any questions or concerns, or If you suspect you are unable to land your PRs for reasons other than the familiar test failures and flakes, please contact any of: @yjbanov, @jtmcdole, @christopherfujino, @Piinks.
```

**Pull request merged**